assignment-1/IPL.java

```diff
@@ -0,0 +1,329 @@
+    }
+
+    public static void getHighestWicketTakerOfTeam(String teamName) {
+        for (Team team : teams) {
+            if (team.teamName.equals(teamName)) {
+                Integer maxTillNow = 0;
+                Player bestBowler = null;
+                for (Player p : team.squad) {
+                    if (p.wickets >= maxTillNow) {
+                        bestBowler = p;
+                        maxTillNow = p.wickets;
+                    }
+                }
+                assert bestBowler != null;
+                bestBowler.display();
+            }
+
+        }
+    }
+
+    public static void getHighestRunScorerOfTeam(String teamName) {
+        for (Team team : teams) {
+            if (team.teamName.equals(teamName)) {
+                Integer maxTillNow = 0;
+                Player bestScorer = null;
+                for (Player p : team.squad) {
+                    if (p.runs >= maxTillNow) {
+                        bestScorer = p;
+                        maxTillNow = p.runs;
+                    }
+                }
+                assert bestScorer != null;
+                bestScorer.display();
+            }
+        }
+    }
+
+    public static void gettingTopPerformers() {
+        ArrayList<Player> top3Batsmen = getTopPlayersByRole("BATSMEN");
+        ArrayList<Player> top3Bowler = getTopPlayersByRole("BOWLER");
+        ArrayList<Player> top3AllRounder = getTopPlayersByRole("ALL ROUNDER");
+        displayArrayList(top3Batsmen);
+    }
+
+    static void displayArrayList(ArrayList<Player> players) {
+        for (Player player : players) {
+            player.display();
+        }
+    }
+
+    public static ArrayList<Player> getTopPlayersByRole(String role) {
+        Stack<Player> top3 = new Stack<>();
+        ArrayList<Player> result = new ArrayList<>();
+        for (Team team : teams) {
+            for (Player p : team.squad) {
+                if (p.role.equals(role)) {
+                    if (top3.size() == 0) {
+                        top3.push(p);
+                    } else {
+                        Stack<Player> temp = new Stack<>();
+                        while (!top3.isEmpty() && top3.peek().runs < p.runs) {
+                            temp.push(top3.pop());
+                        }
+                        temp.push(p);
+                        while (!temp.isEmpty()) {
+                            top3.push(temp.pop());
+                        }
+                        while (top3.size() > 3) {
+                            top3.pop();
+                        }
+                    }
+                }
+            }
+        }
+        while (!top3.isEmpty()) {
+            result.add(top3.pop());
+        }
+        return result;
+    }
+
+    public static void bestPossiblePerformance() {
+        HashMap<Team, Double> mapTeamToPredictedScore = new HashMap<>();
+        for (Team team : teams) {
+            Comparator<Player> cmp = Comparator.comparing(Player::getRuns);
```

Comment:
Comparison should have been based on the average and not the total runs

Reply:
I had doubt if some player has played very few match, he would be having much higher average, which is unreal so tried to filter out based on total runs.